Sources/TensorFlow/Operators/Math.swift

```diff
@@ -2317,12 +2317,24 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
     /// dimensions are removed.
```

Comment:
Add a precondition which states that `axes.rank` must be `0`.

Reply:
It has to be rank `1`, but I didn't add it because it would cause lazy tensor materialization. The same is true for all reduction ops and we don't use any preconditions there.

Reply:
Oh sorry I meant rank `1`.  Precondition should be added in comments at least.

Reply:
Good point. I added it in the documentation strings.